### PR TITLE
fix std in DataMisfit and possible bug in PlotUtils

### PR DESCRIPTION
**Pull request by thast**
Fix for Issue #746

- fixed `DataMisfit.std` that now take into consideration `survey.std`. So you are not forced to set `DataMisfit.W` manually anymore

-  Fixed: `Survey.MakeSyntheticData`, if not given a `std`, was overwriting `survey.std` even if already set.  Now it will use `survey.std` if not given a `std` argument.

Explanation:
`DataMisfit.std` was set to 5% by default, but to change it to `survey.std`, it expected a `None` value, thus it never gets updated...

- bonus: light improvement again of `Plot2DData` (reordering of `if` statement so specifying `contourOpts['vmin','vmax']`  without a `clim` works.

- bonus: Somehow PR #725 did not make it to improve `plot_misfit_curves` so I put it back here. Mainly, more option to plot $\phi_m$ or $\phi_small$  and $\phi_smooth$ independently, plus more options to save the figure.

### CODE REVIEW

**Comment by micmitch** [comment]
Your changes to DataMisfit.std look good to me.

**Comment by thast** [comment]
@lheagy : any idea what happened with PR #725 ? It seems the changes made in `Directives` did not make it in Master even so we merged the PR (and I do not see anything in the history).
I'm curious and I do not understand how that could have happened.

**Comment by lheagy** [comment]
Thanks @thast: don't worry about travis yet... since you updated a computationally-expensive example, we will need to send the updates to the google cloud storage in order for it to pass (so feel free to revert the changes). Would you mind including a small example of the previous (buggy) behaviour and how this fixes it? This would also make a good addition to the tests so that we don't break the expected behaviour in the future.

**Comment by thast** [comment]
@lheagy : Thanks. I will revert back Travis. I do not know when I will be able to provide a test and example though but will do my best to get it quickly.

File: SimPEG/DataMisfit.py [hunk 1]
```diff
@@ -58,19 +58,20 @@ class l2_DataMisfit(BaseDataMisfit):
...
+    std = None  #: default standard deviation if not provided by survey
     eps = None  #: default floor
     eps_factor = 1e-5  #: factor to multiply by the norm of the data to create floor
 
     def __init__(self, survey, **kwargs):
         BaseDataMisfit.__init__(self, survey, **kwargs)
 
         if self.std is None:
-            if getattr(self.survey, 'std', None) is not None:
+            if getattr(self.survey, 'std', None) is None:
```
Comment (lheagy) [inline SimPEG/DataMisfit.py]: Thanks for catching this @thast. This won't yet catch the case if the user changes the standard deviation on the survey after the data misfit has been created. I think that is okay for now, but lets leave an issue open and we can work to address this with the simulation refactor.

**Comment by thast** [comment]
Hi @lheagy , I chose to go a slightly different way, building the test on already existing tests for DataMisfit.

Also, I've found that the function MakeSyntheticData, if not given a std, would also overwrite survey.std even if already set. So I've fixed it to use it survey.std if not given a std argument.

File: tests/base/test_DataMisfit.py [hunk 2]
```diff
@@ -41,6 +44,8 @@ def setUp(self):
         self.prob = prob
         self.dobs = dobs
         self.dmis = dmis
+        import SimPEG
```
Comment (lheagy) [inline tests/base/test_DataMisfit.py]: is this used in the test at all?

Comment (thast) [reply tests/base/test_DataMisfit.py]: ah good catch. I was experimenting something while writing it and forgot to remove some part of that experimentation....

Comment (thast) [reply tests/base/test_DataMisfit.py]: fixed

**Comment by thast** [comment]
Good job with travis @lheagy!

**Pull request merged**